File: RxDemo/src/main/java/com/vondear/rxdemo/activity/ActivityMovieSeat.java
package com.vondear.rxdemo.activity;

import android.os.Bundle;
import android.widget.LinearLayout;

import com.vondear.rxdemo.R;
import com.vondear.rxtool.RxBarTool;
import com.vondear.rxtool.RxDeviceTool;
import com.vondear.rxui.activity.ActivityBase;
import com.vondear.rxui.view.RxSeatMovie;
import com.vondear.rxui.view.RxTitle;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author vondear
 */
public class ActivityMovieSeat extends ActivityBase {

    @BindView(R.id.seatView)
    RxSeatMovie mSeatView;
    @BindView(R.id.activity_movie_seat)
    LinearLayout mActivityMovieSeat;
    @BindView(R.id.rx_title)
    RxTitle mRxTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        RxBarTool.noTitle(this);
        RxBarTool.setTransparentStatusBar(this);
        setContentView(R.layout.activity_movie_seat);
        ButterKnife.bind(this);
        RxDeviceTool.setPortrait(this);
        initView();
    }


    protected void initView() {
        mRxTitle.setLeftFinish(mContext);

        mSeatView.setScreenName("3号厅荧幕");//设置屏幕名称
        mSeatView.setMaxSelected(8);//设置最多选中

        mSeatView.setSeatChecker(new RxSeatMovie.SeatChecker() {

            @Override
            public boolean isValidSeat(int row, int column) {
                return !(column == 2 || column == 12);
            }

            @Override
            public boolean isSold(int row, int column) {
                return row == 6 && column == 6;
            }

            @Override
            public void checked(int row, int column) {

            }

            @Override
            public void unCheck(int row, int column) {

            }

            @Override
            public String[] checkedSeatTxt(int row, int column) {
                return null;
            }

        });
        mSeatView.setData(10, 15);
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityMovieSeat.kt
package com.tamsiree.rxdemo.activity

import android.os.Bundle
import com.tamsiree.rxdemo.R
import com.tamsiree.rxkit.RxBarTool.noTitle
import com.tamsiree.rxkit.RxBarTool.setTransparentStatusBar
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxui.activity.ActivityBase
import com.tamsiree.rxui.view.RxSeatMovie.SeatChecker
import kotlinx.android.synthetic.main.activity_movie_seat.*

/**
 * @author tamsiree
 */
class ActivityMovieSeat : ActivityBase() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        noTitle(this)
        setTransparentStatusBar(this)
        setContentView(R.layout.activity_movie_seat)
        setPortrait(this)
    }

    override fun initView() {
        rx_title.setLeftFinish(mContext)
        seatView.setScreenName("3号厅荧幕") //设置屏幕名称
        seatView.setMaxSelected(8) //设置最多选中
        seatView.setSeatChecker(object : SeatChecker {
            override fun isValidSeat(row: Int, column: Int): Boolean {
                return !(column == 2 || column == 12)
            }

            override fun isSold(row: Int, column: Int): Boolean {
                return row == 6 && column == 6
            }

            override fun checked(row: Int, column: Int) {}
            override fun unCheck(row: Int, column: Int) {}
            override fun checkedSeatTxt(row: Int, column: Int): Array<String>? {
                return null
            }
        })
        seatView.setData(10, 15)
    }

    override fun initData() {

    }
}